backend/src/agent/persistence/client/token_refresher.go

```diff
@@ -0,0 +1,78 @@
+package client
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"sync"
+	"time"
+)
+
+type TokenRefresherInterface interface {
+	GetToken() string
+	RefreshToken() error
+}
+
+const SaTokenFile = "/var/run/secrets/kubeflow/tokens/persistenceagent-sa-token"
+
+type FileReader interface {
+	ReadFile(filename string) ([]byte, error)
+}
+
+type tokenRefresher struct {
+	mu         sync.RWMutex
+	seconds    *time.Duration
+	token      string
+	fileReader *FileReader
+}
+
+type FileReaderImpl struct{}
+
+func (r *FileReaderImpl) ReadFile(filename string) ([]byte, error) {
+	return os.ReadFile(filename)
+}
+
+func NewTokenRefresher(seconds time.Duration, fileReader FileReader) *tokenRefresher {
+	if fileReader == nil {
+		fileReader = &FileReaderImpl{}
+	}
+
+	tokenRefresher := &tokenRefresher{
+		seconds:    &seconds,
+		fileReader: &fileReader,
+	}
+
+	return tokenRefresher
+}
+
+func (tr *tokenRefresher) StartTokenRefreshTicker() error {
+	err := tr.RefreshToken()
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(*tr.seconds)
+	go func() {
+		for range ticker.C {
```

Comment:
Another thing, I was unsure if I should use the `stopCh`, so I removed it. WDYT?  Please inspect the changes introduced by my third [commit](https://github.com/kubeflow/pipelines/pull/9699/commits/bc9e1e05d35127a0ce8e5fd1e92a3b5e85ca9ac6).